src/creature.cpp

```diff
@@ -192,6 +192,95 @@ void Creature::setpos( const tripoint &p )
     on_move( old_loc );
 }
 
+static units::volume size_to_volume( creature_size size_class )
+{
+    // returns midpoint of size from volume_to_size, minus 1_ml
+    // e.g. max tiny size is 7500, max small size is 46250, we return
+    // 46250+7500 / 2 - 1_ml = 26875_ml - 1ml
+    // This is still stupid and both of these functions should be merged into one single source of truth.
+    if( size_class == creature_size::tiny ) {
+        return 3749_ml;
+    } else if( size_class == creature_size::small ) {
+        return 26874_ml;
+    } else if( size_class == creature_size::medium ) {
+        return 77124_ml;
+    } else if( size_class == creature_size::large ) {
+        return 295874_ml;
+    }
+    return 741874_ml;
+}
+
+bool Creature::can_move_to_vehicle_tile( const tripoint_abs_ms &loc, bool &cramped ) const
+{
+    map &here = get_map();
+    const optional_vpart_position vp_there = here.veh_at( loc );
+    if( !vp_there ) {
+        return true;
+    }
+
+    const monster *mon = as_monster();
+
+    vehicle &veh = vp_there->vehicle();
+
+    std::vector<vehicle_part *> cargo_parts;
+    cargo_parts = veh.get_parts_at( here.bub_from_abs( loc ), "CARGO", part_status_flag::any );
+
+    units::volume capacity = 0_ml;
+    units::volume free_cargo = 0_ml;
+    for( vehicle_part *part : cargo_parts ) {
+        vehicle_stack contents = veh.get_items( *part );
+        if( !vp_there.part_with_feature( "CARGO_PASSABLE", false ) &&
+            !vp_there.part_with_feature( "APPLIANCE", false ) &&
+            !vp_there.part_with_feature( "OBSTACLE", false ) ) {
+            capacity += contents.max_volume();
+            free_cargo += contents.free_volume();
+        }
+    }
+    if( capacity > 0_ml ) {
+        // First, we'll try to squeeze in. Open-topped vehicle parts have more room to step over cargo.
+        if( !veh.enclosed_at( here.getlocal( loc ) ) ) {
+            free_cargo *= 1.2;
+        }
+        const creature_size size = get_size();
+        units::volume critter_volume;
+        if( mon ) {
+            critter_volume = mon->get_volume();
+        } else {
+            critter_volume = size_to_volume( size );
+        }
+
+        if( critter_volume > free_cargo ) {
+            return false;
+        }
+
+        if( critter_volume > size_to_volume( creature_size::large ) &&
+            !vp_there.part_with_feature( "HUGE_OK", false ) ) {
+            return false;
+        }
+
+        if( critter_volume < free_cargo * 1.33 ) {
```

Comment:
Am I reading this right? If the creature's volume is *less* than a third over the free cargo space and isn't a monster, snake, blob or fish-type then it's cramped? Shouldn't the check be for if the creature's volume *greater* than a third over free cargo space?

Reply:
Reading it right, interpreting it wrong. If the critter is greater than the free cargo size, it can't move into this tile in the first place - we return false earlier in the function. https://github.com/CleverRaven/Cataclysm-DDA/pull/74004/files#diff-5a026da090b6664f91c5f3bc4a9d0844cff966ef6fa5785d258b3aa47b370a2cR253

But yeah this logic is confusing, and part of the reason why I wanted to get rid of the huge block of volume comparisons shown.

Reply:
Ah, yeah, point. I was so focused on chasing down what was setting cramped to true I failed to remember what the whole function was about. 

That said, I'm reasonably sure this particular check is the culprit when it comes to #74068 since commenting out `cramped = true;` and recompiling prevents the behavior reported there from occurring.

Reply:
Yes, the equation should be `free_cargo * 0.75 < critter_volume` instead,
thus we set cramped when `free_cargo * 0.75 < critter_volume && critter_volume <= free_cargo`

I'm already working on it tonight, this whole thing is still a mess.

Reply:
Just tested `critter_volume * 0.75 < free_cargo  &&  critter_volume <= free_cargo` and it's still setting cramped to true.